controllers/managedocs_controller.go

```diff
@@ -604,44 +667,71 @@ func (r *ManagedOCSReconciler) reconcileAlertmanagerConfigSecret() error {
+func (r *ManagedOCSReconciler) generateAlertmanagerConfig(pagerdutyServiceKey string, dmsURL string) interface{} {
+	type PagerDutyConfig struct {
+		ServiceKey string `yaml:"service_key"`
+	}
+
+	type WebhookConfig struct {
+		Url string `yaml:"url"`
+	}
 
 	alertmanagerConfig := struct {
 		Route struct {
-			GroupWait      string `yaml:"group_wait"`
-			GroupInterval  string `yaml:"group_interval"`
-			RepeatInterval string `yaml:"repeat_interval"`
-			Receiver       string `yaml:"receiver"`
-			Routes         [1]struct {
-				Match struct {
-					Severity string `yaml:"severity"`
-				} `yaml:"match"`
-				Receiver string `yaml:"receiver"`
+			GroupWait      string `yaml:"group_wait,omitempty"`
+			GroupInterval  string `yaml:"group_interval,omitempty"`
+			RepeatInterval string `yaml:"repeat_interval,omitempty"`
+			Receiver       string `yaml:"receiver,omitempty"`
+			Routes         [2]struct {
+				GroupWait      string            `yaml:"group_wait,omitempty"`
+				GroupInterval  string            `yaml:"group_interval,omitempty"`
+				RepeatInterval string            `yaml:"repeat_interval,omitempty"`
+				Receiver       string            `yaml:"receiver,omitempty"`
+				Match          map[string]string `yaml:"match,omitempty"`
 			} `yaml:"routes"`
 		} `yaml:"route"`
-		Receivers [1]struct {
-			Name             string `yaml:"name"`
-			PagerdutyConfigs [1]struct {
-				ServiceKey string `yaml:"service_key"`
-			} `yaml:"pagerduty_configs"`
+		Receivers [3]struct {
+			Name             string            `yaml:"name"`
+			PagerdutyConfigs []PagerDutyConfig `yaml:"pagerduty_configs,omitempty"`
+			WebhookConfigs   []WebhookConfig   `yaml:"webhook_configs,omitempty"`
 		} `yaml:"receivers"`
 	}{}
 
-	alertmanagerConfig.Route.GroupWait = "30s"
-	alertmanagerConfig.Route.GroupInterval = "5m"
-	alertmanagerConfig.Route.RepeatInterval = "4h"
 	alertmanagerConfig.Route.Receiver = "pagerduty"
-	alertmanagerConfig.Route.Routes[0].Match.Severity = "critical"
+
+	alertmanagerConfig.Route.Routes[0].GroupWait = "30s"
+	alertmanagerConfig.Route.Routes[0].GroupInterval = "5m"
+	alertmanagerConfig.Route.Routes[0].RepeatInterval = "4h"
 	alertmanagerConfig.Route.Routes[0].Receiver = "pagerduty"
+	alertmanagerConfig.Route.Routes[0].Match = make(map[string]string)
+	alertmanagerConfig.Route.Routes[0].Match["severity"] = "critical"
+
+	alertmanagerConfig.Route.Routes[1].GroupWait = "5m"
+	alertmanagerConfig.Route.Routes[1].GroupInterval = "5m"
+	alertmanagerConfig.Route.Routes[1].RepeatInterval = "5m"
+	alertmanagerConfig.Route.Routes[1].Receiver = "deadmanssnitch"
+	alertmanagerConfig.Route.Routes[1].Match = make(map[string]string)
+	alertmanagerConfig.Route.Routes[1].Match["alertname"] = "deadmanssnitch"
+
 	alertmanagerConfig.Receivers[0].Name = "pagerduty"
-	alertmanagerConfig.Receivers[0].PagerdutyConfigs[0].ServiceKey = pagerdutyServiceKey
+
+	alertmanagerConfig.Receivers[1].Name = "pagerduty"
```

Comment:
This receiver has the same name as the one in position 0 (which has not configuration attached), do we really need 3 receivers? if so, we cannot name them the same name (the name is used by the route to identify a receiver).

Reply:
This was a mistake on my part, I'll change it back to two.